pkg/agent/openflow/pipeline.go

```diff
@@ -243,44 +243,69 @@ const (
 	// macRewriteMark indicates the destination and source MACs of the
 	// packet should be rewritten in the l3ForwardingTable.
 	macRewriteMark = 0b1
-	cnpDropMark    = 0b1
+	// cnpDenyMark indicates the packet is denied(Drop/Reject).
+	cnpDenyMark = 0b1
 
 	// gatewayCTMark is used to to mark connections initiated through the host gateway interface
 	// (i.e. for which the first packet of the connection was received through the gateway).
 	gatewayCTMark = 0x20
 	snatCTMark    = 0x40
 	ServiceCTMark = 0x21
 
-	// disposition is loaded in marksReg [21]
+	// disposition is loaded in marksReg [21-22]
 	DispositionMarkReg regType = 0
-	// disposition marks the flow action as either Allow or Drop
-	DispositionAllow = 0b0
-	DispositionDrop  = 0b1
+	// disposition marks the flow action
+	DispositionAllow = 0b00
+	DispositionDrop  = 0b01
+	DispositionRej   = 0b10
+
+	// custom reason is loaded in marksReg [24-25]
+	CustomReasonMarkReg regType = 0
+	// custom reason marks the reason of sending packet to the controller, which could
+	// be added up to implement a multi-reason combination.
+	// e.g. 0b11 (CustomReasonLogging + CustomReasonReject) means both Logging and
+	// Reject are reasons of sending packet to the controller
```

Comment:
This comment should be moved up to `CustomReasonMarkReg` IMO, it doesn't make sense for it to be a comment for `CustomReasonLogging`. `CustomReasonLogging` should have its own comment, just like `CustomReasonReject` has one.

Reply:
Changed.